addons/website_sale/controllers/main.py

```diff
@@ -150,9 +150,10 @@ def _get_pricelist_context(self):
         return pricelist_context, pricelist
 
     def _get_compute_currency(self, pricelist, product=None):
-        from_currency = (product or request.env.user.company_id).currency_id
+        company = product and product._get_current_company(pricelist=pricelist, website=request.website) or pricelist.company_id or request.website.company_id
+        from_currency = (product or request.env['res.company']._get_main_company()).currency_id
```

Comment:
Shouldn't this be website company?

Reply:
As explained on the 2nd commit, when using multi-company, the base currency with which all prices are encoded is the one from the main company, and not contextual.

What is contextual is the currency rate.

Reply:
OK, thanks for explaining.

Reply:
You're welcome. If you want the full details, the only place where this method is used is list view of variants to convert the `lst_price` of the variant (encoded in base currency) to the website currency.

Eventually I want to get rid of that method `_get_compute_currency` all together but this is for master, in all the other places in 12.0 except that one it has already been replaced by `get_combination_info`.

Reply:
Thanks also for the expanded details!! All the price computation through pricelist requires a bit refactoring IMO. It's hardly extendable (look for example the ugly hacks for getting some things in https://github.com/OCA/product-attribute/tree/11.0/product_pricelist_supplierinfo).

Ping me if you want me to review anything for master regarding this.